cmd/clusterctl/client/init.go

```diff
@@ -60,6 +60,10 @@ type InitOptions struct {
 
 	// LogUsageInstructions instructs the init command to print the usage instructions in case of first run.
 	LogUsageInstructions bool
+
+	// SkipVariables skips variable parsing in the provider components yaml.
+	// It is set to true for listing images of provider components.
+	SkipVariables bool
```

Comment:
@wfernandes I'm s little bit concerned about adding this field to the InitOptions struct, because this gives to the library users the possibility to do init with skip options that is something we don't want.
Have you considered instead to add an additional arg to the private method setupInstaller:

```
func (c *clusterctlClient) setupInstaller(cluster cluster.Client, options InitOptions, SkipVariables bool) (cluster.ProviderInstaller, error)
```

Reply:
Good catch. I agree that it need not be exported. 
I'm not too stuck on the implementation but another suggestion is just making the `skipVariables bool` private and update the comment to say it is meant for `InitImages`.
That way if we ever do split up the `InitImages` into a separate command, it can be easily included as an InitImageOption.
Again, not too hung up on the implementation, it's just another suggestion. WDYT?

Reply:
ok for me